### PR TITLE
fix: TS program doesn't emit declaration files

### PR DESCRIPTION
**Pull request by pavandv**
This PR addresses 3 main issues:

1. TS doesn't emit declaration files. TS isn't able to resolve the Source files passed to `ts.createProgram`, from `federationOptions#exposes`. The file paths mentioned in `exposes` may not always be relative to the `cwd()`.
2. `__types_index.json` has wrong file paths that aren't in sync with the emitted `.d.ts` files.
3. When downloading Remote Types, we always check for `@` (app1@http://localhost:3000) in the remote url defined. This condition shouldn't necessarily be true always.

### CODE REVIEW

File: src/index.js
Comment (vadym-oliinyk) [inline src/index.js:51]: Looks like here should be `.ts`

Comment (ScriptedAlchemy) [reply src/index.js:51]: Good find

File: src/index.js
Comment (ScriptedAlchemy) [inline src/index.js:71]: What happens if theres multiple dots like thing.component.tsx. Path.basename or path.parse may be more reliable

File: src/index.js
Comment (pavandv) [reply src/index.js]: addressed it. 👍

File: src/index.js
Comment (pavandv) [reply src/index.js]: We cannot actually use this map() as the ts compiler emits the filenames as absolute paths.

addressed this with a different approach.

**Review by ScriptedAlchemy — COMMENTED** [review]
Will review again in a few hours. Once approved, feel free to merge when ready and I'll cut a new release

**Comment by joshuafonseca** [comment]
I have tried the checked in code. When integrating with webpack and angular, the clean up plugin is removing the @mf-typescript folder. It will be good to make use of webpack hooks to avoid these kind of issues (I have used the done hook, however, I'm not sure what should be the correct hook) and emit the types as assets. Also, the __dirname variable is pointing to the package's folder. So, I have resolved the context from the module's webpack.config.js and then referred to it like this.compilerOptions.context

**Review by ScriptedAlchemy — APPROVED** [review]

**Comment by pavandv** [comment]
@joshuafonseca Can you please create a new issue and provide as a sample repo to reproduce the issue you are describing. Thank you!!

**Pull request merged**
